### PR TITLE
Implement task processing queue collection

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
Implement task processing queue collection, which manages a set of non-overlapping processing queues

**Why?**
Part of multi-cursor queue implementation

**How did you test it?**
Unit test

**Potential risks**
N/A, implementation not used yet.

### CODE REVIEW

File: service/history/queue/processing_queue.go [hunk 1]
```diff
@@ -221,19 +221,25 @@ func (q *processingQueueImpl) Merge(
...
-	newQueueStates = append(newQueueStates, newProcessingQueueState(
+	overlappingQueueState := newProcessingQueueState(
 		q1.state.level,
 		maxTaskKey(q1.state.ackLevel, q2.state.ackLevel),
 		minTaskKey(q1.state.readLevel, q2.state.readLevel),
 		minTaskKey(q1.state.maxLevel, q2.state.maxLevel),
 		q1.state.domainFilter.Merge(q2.state.domainFilter),
-	))
+	)
+	if overlappingQueueState.readLevel.Less(overlappingQueueState.ackLevel) {
```
Comment (yux0) [inline service/history/queue/processing_queue.go:231]: should this handle during initializing the queue state?

File: service/history/queue/processing_queue_collection.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+)
+
+// NewProcessingQueueCollection creates a new collection for non-overlapping queues
+func NewProcessingQueueCollection(
+	level int,
+	queues []ProcessingQueue,
+) ProcessingQueueCollection {
+	sortProcessingQueue(queues)
+	queueCollection := &processingQueueCollection{
+		level:  level,
```
Comment (yux0) [inline service/history/queue/processing_queue_collection.go:44]: What does this level mean?

File: service/history/queue/processing_queue.go (hunk 1 above)
Comment (yycptt) [reply service/history/queue/processing_queue.go]: Updated. Moved the logic into queue state initialization.

File: service/history/queue/processing_queue_collection.go (hunk 2 above)
Comment (yycptt) [reply service/history/queue/processing_queue_collection.go:44]: Consider it as the collectionID. All the queues inside a queue collection will have the same `level` which equals to the level of the collection (collectionID).

**Pull request merged**